apps/angular/projection/src/app/component/teacher-card/teacher-card.component.ts

```diff
@@ -21,20 +36,22 @@ import { CardComponent } from '../../ui/card/card.component';
     `,
   ],
   standalone: true,
-  imports: [CardComponent],
+  imports: [CardComponent, AsyncPipe, ListItemComponent],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class TeacherCardComponent implements OnInit {
-  teachers: Teacher[] = [];
-  cardType = CardType.TEACHER;
+  private store = inject(TeacherStore);
+  private http = inject(FakeHttpService);
 
-  constructor(
-    private http: FakeHttpService,
-    private store: TeacherStore,
-  ) {}
+  protected teachers$ = this.store.teachers$;
```

Comment:
use signal to bind your properties to the template. You will get all future benefit and it's good to get used to it